src/sql/src/session/vars.rs

```diff
@@ -2871,52 +2871,16 @@ impl Value for Vec<Ident> {
     }
 }
 
-impl Value for Option<String> {
+// Implement `Value` for `Option<V>` for any owned `V`.
+impl<V: Value + Clone + ToOwned<Owned = V>> Value for Option<V> {
+    // `Option<V>` value parameters require string LaunchDarkly types.
     const TYPE_NAME: &'static str = "optional string";
```

Comment:
This seems funny! The type name in Materialize should have as much precision as possible, even if we have to cheat a bit in LD.

```suggestion
    const TYPE_NAME: &'static str = concat!("optional ", V::TYPE_NAME);
```

Reply:
Note that the `TYPE_NAME` also shows up for non-private `Option<T>` session / server vars when you type `SHOW ALL` as a user.

Reply:
Oh i forgot about `concat!`!!

Reply:
```
error[E0401]: can't use generic parameters from outer function
    --> src/sql/src/session/vars.rs:2909:67
     |
2905 | impl<V> Value for Option<V>
     |      - type parameter from outer function
...
2909 |     const TYPE_NAME: &'static str = concatcp!("optional string", <V as Value>::TYPE_NAME);
     |                                                                   ^ use of generic parameter from outer function

note: erroneous constant used
    --> src/sql/src/session/vars.rs:2909:37
     |
2909 |     const TYPE_NAME: &'static str = concatcp!("optional string", <V as Value>::TYPE_NAME);
     |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: this note originates in the macro `$crate::pmr::__concatcp_impl` which comes from the expansion of the macro `concatcp` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant used
    --> src/sql/src/session/vars.rs:2909:37
     |
2909 |     const TYPE_NAME: &'static str = concatcp!("optional string", <V as Value>::TYPE_NAME);
     |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: this note originates in the macro `__cf_osRcTFl4A::__concatcp_inner` which comes from the expansion of the macro `concatcp` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0401`.
error: could not compile `mz-sql` due to previous error
warning: build failed, waiting for other jobs to finish...
    Building [=======================> ] 771/776: clusterd(bin)
    
  
```

its unclear to me if this possible at all in rust right now :(